src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -2043,15 +2001,12 @@ static void recv_trailing_metadata_ready_for_retries(
   // Not retrying, so commit the call.
   retry_commit(elem, retry_state);
   // Now that the try is committed, give the trailer to the lb policy as needed
```

Comment:
This means the lb recv_trailing_metadata_ready callback can be called more than once. Do we need some way to notify the lb when the call is finally committed?
At the very least, I think we might need to change it to GRPC_CLOSURE_RUN() to run the callback synchronously at the top, because it seems like the retries code path owns the metadata and may modify it before the lb callback gets a chance to run.